### PR TITLE
Wait for internal db on startup and always print error message on reql connection failure

### PR DESCRIPTION
**Pull request by Tryneus**
As part of the fix for #639, these changes should make it more obvious when there is a connection problem to the database.  Reorganized some of the code in the `Metadata` constructor to be a more linear promise chain as part of this.

This does not address all aspects of #639, as these is still no mechanism for passing errors to the user of the `Server` object when there is connectivity problems with the database - it is only reported through the logger.

---

This change is [Reviewable](https://reviewable.io/reviews/rethinkdb/horizon/644)

### CODE REVIEW

**Comment by casche** [comment]
tl;dr;

Hey @Tryneus thanks again for looking at this 👍. Additionally, it would be helpful in this case to also receive positive feedback when Horizon connects on startup in embedded mode. 

for example:

```
info: Connecting to RethinkDB: localhost:28015
```

then

```
info: Connected to RethinkDB: localhost:28015

```

In my architecture, I have the horizon application and rethinkdb in their own containers. When they start I only receive the feedback that it is connecting. I never know it 'connected'. 

## 

Long version 😆 

server.js

``` javascript
const app = express();

app.get('*', function(req, res) {
  res.sendFile(path.join( __dirname, '../src/index.html'));
});

const httpServer = app.listen(appConfig.port, function(err) {
  if (err) {
    console.log(err);
  } else {
    open(`http://localhost:${appConfig.port}`);
  }
});

const horizonServer = horizon(httpServer, {
   auto_create_collection: true,
   auto_create_index: true,
   project_name: appConfig.title,
   permissions: false,
   auth: {
     allow_anonymous: true,
     allow_unauthenticated: true,
     token_secret: appConfig.token_secret
   }
 });

 export default httpServer;
```

docker-compose.yml

``` yml
version: "2"
services:
  web:
    build: .
    depends_on:
      - db
    ports:
      - "80:8080"
  db:
    image: rethinkdb
    ports:
      - "28015:28015"
```

As you might expect the docker container just runs `npm start` which runs `node server.js`. In this setup I never know if the connection occurred:

``` bash
$ docker-compose up
Creating network "beauty_default" with the default driver
Recreating beauty_db_1
Recreating beauty_web_1
Attaching to beauty_db_1, beauty_web_1
web_1  | 
web_1  | > lightbrew-beauty@0.0.1 prestart /usr/src/app
web_1  | > babel-node tools/startMessage.js
web_1  | 
db_1   | Running rethinkdb 2.3.4~0jessie (GCC 4.9.2)...
db_1   | Running on Linux 4.4.12-boot2docker x86_64
db_1   | Loading data from directory /data/rethinkdb_data
db_1   | warn: Cache size does not leave much memory for server and query overhead (available memory: 680 MB).
db_1   | warn: Cache size is very low and may impact performance.
db_1   | Listening for intracluster connections on port 29015
db_1   | Listening for client driver connections on port 28015
db_1   | Listening for administrative HTTP connections on port 8080
web_1  | Starting app in dev mode...
web_1  | 
web_1  | > lightbrew-beauty@0.0.1 start /usr/src/app
web_1  | > node tools/distServer.js
web_1  | 
db_1   | Listening on cluster addresses: 127.0.0.1, 172.18.0.2, ::1, fe80::42:acff:fe12:2%39
db_1   | Listening on driver addresses: 127.0.0.1, 172.18.0.2, ::1, fe80::42:acff:fe12:2%39
web_1  | info: Connecting to RethinkDB: localhost:28015
db_1   | Listening on http addresses: 127.0.0.1, 172.18.0.2, ::1, fe80::42:acff:fe12:2%39
db_1   | Server ready, "2a361cf80bfd_wur" 51785a58-556a-45d4-9277-1b54b430fdc8
```

File: server/src/metadata/metadata.js [hunk 1]
```diff
@@ -201,35 +228,8 @@ class Metadata {
                 }
               }).catch(reject);
             });
-          });
-      return Promise.all([ groups_ready, collections_ready, indexes_ready ]);
```
Comment (deontologician) [inline server/src/metadata/metadata.js:205]: This syntax of giving them intermediate variables with meaningful names seems better than the new version

**Comment by deontologician** [comment]
Other than the one comment about clarity, lgtm :+1:

**Comment by Tryneus** [comment]
@deontologician - I added back in the changefeed variable names for clarity.

@casche - we already had a log statement for that (`Metadata synced with database, ready for traffic.`), but I changed it to `Connection to RethinkDB ready: user @ host:port` similar to your suggestion.  If you want to check for readiness programmatically, you should use the `server.ready()` promise.  This will be reset to a new Promise every time we lose the connection to the database - `server.is_ready()` should indicate if that is necessary.

**Comment by casche** [comment]
@Tryneus 👍 ⭐ :shipit:
